dkg/pedersen/handler.go

```diff
@@ -26,13 +26,57 @@ import (
 // Allows to exit the loop.
 const recvTimeout = time.Second * 4
 
-// the time after which we expect new messages (deals or responses) to be
-// received.
-const retryTimeout = time.Millisecond * 200
+// the channel size used to stored buffered deals and responses. This arbitrary
+// value must be set according to the characteristic of the system: number of
+// nodes, networking, memory, etc...
+const chanSize = 200
 
 // constant used in the logs
 const newState = "new state"
 
+func newCryChan[T any](bufSize int) cryChan[T] {
+	llvl := zerolog.NoLevel
+	if os.Getenv("CRY_LVL") == "warn" {
+		llvl = zerolog.WarnLevel
+	}
+
+	return cryChan[T]{
+		c: make(chan T, bufSize),
+		log: dela.Logger.With().Str("role", "cry chan").Int("size", bufSize).
+			Logger().Level(llvl),
+	}
+}
+
+type cryChan[T any] struct {
+	c   chan T
+	log zerolog.Logger
+}
+
+func (c *cryChan[T]) push(e T) {
+	start := time.Now()
+	select {
+	case c.c <- e:
+	case <-time.After(time.Millisecond * 100):
+		c.log.Warn().Str("obj", fmt.Sprintf("%+v", e)).Msg("channel blocking")
+		c.c <- e
+		c.log.Warn().Str("obj", fmt.Sprintf("%+v", e)).
+			Str("elapsed", time.Since(start).String()).Msg("channel unblocked")
+	}
+}
+
+func (c *cryChan[T]) pop(ctx context.Context) (t T, err error) {
+	select {
+	case el := <-c.c:
+		return el, nil
+	case <-ctx.Done():
+		return t, ctx.Err()
+	}
+}
```

Comment:
when we move this out, I'd have a few comments regarding API design decisions that I'd change. For this PR, this is good enough.

Writing down my (as-of-yet-incoherent) thoughts so I don't forget later on: 
- I'd consider exposing the channel directly for reading, as a `context` might not always be flexible enough
- I wonder if we should set a global `write timeout` to the channel (much like `SetWriteDeadline`), or be able to provide individual timeouts for writes
- The "crying" should probably provide a unique identifier (say, a go routine ID) so that we can make sense of logs and distinguish between multiple unique blocked channels.